### DISCUSSION TITLE
Plex not updating

### DISCUSSION THREAD

**Discussion by nmbrg** [tteck/Proxmox#2520] [Help Needed] [answered]
im trying to update to the latest plex server version (1.40.0.7998) with the command provided at: https://tteck.github.io/Proxmox/

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/plex.sh)"
(note that i have an older version of the script where the 'update' command wasnt available)

and it seems to be ok.
![image](https://github.com/tteck/Proxmox/assets/31845361/724ed89f-fdde-42cc-8da5-e77b079e0da4)

But the plex server isnt updated.
i also tried rebooting the LXC in the hopes of it checking for newer versions and updating when booting up, but perhaps that isnt a thing. Because it didnt.

any ideas on why? what can i do instead to get it updated?

**Comment by tteck** [comment]
Plex Media Server is updated through the standard `apt update && apt upgrade` commands, leveraging its dedicated repository.
https://github.com/tteck/Proxmox/blob/25a9708fe3baca8771d3bb5c7afe210a86a6f121/install/plex-install.sh#L40-L43
Whenever Plex releases an updated `.deb` file to this repository, you can utilize the update command to fetch and install the latest version of Plex Media Server.

**Reply by nmbrg** [reply]
Tried that as well (forgot to mention it). But it says 0 stuff needs updating and nothing is updated.

**Reply by tteck** [reply]
That indicates that Plex hasn't released an updated `.deb` file to their repository.

**Reply by nmbrg** [reply]
It was released to everyone 7 days ago. So seems odd if they haven’t.

**Reply by tteck** [reply]
🤷🏻‍♂️  You can download and install it manually from their website if you're impatient.

**Reply by nmbrg** [reply]
Wouldn’t even know where to start on how to update it manually on a proxmox LXC. So guess I have to wait. But I’m not convinced that’s the issue. They have always had it available at day one of release. And now it’s been 7 days. I guess only time will tell. :)

but I noticed they wrote in the description when checking for new version in the plex web-ui:

“This update will need to be installed manually after download.”

but I’m guessing thats for when updating thru the web-ui. (It usually is)

**Reply by tteck** [reply]
A fresh install shows that the new version **is** in their repository.
```
root@plex:~# dpkg -l | grep plexmediaserver
ii  plexmediaserver               1.40.0.7998-c29d4c0c8                   amd64        Plex organizes all of your personal media so you can easily access and enjoy it.
```

**Reply by nmbrg** [reply]
Then we’re back at my original question. Why isn’t it updating with the commands it should be updating with?

**Reply by tteck** [reply]
show the output of `cat /etc/apt/sources.list.d/plexmediaserver.list`

**Reply by tteck** [reply]
Also, show the output of `dpkg -l | grep plexmediaserver`

**Reply by nmbrg** [reply]
```
root@plex:~# dpkg -l | grep plexmediaserver
ii  plexmediaserver               1.40.0.7775-456fbaf97                   amd64        Plex organizes all of your personal media so you can easily access and enjoy it.
```

and an 'apt update' shows this:
```
root@plex:~# apt update
Hit:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Get:2 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Fetched 229 kB in 1s (269 kB/s)  
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
```
well. it showed 0 new...

heres the apt upgrade:

```
root@plex:~# apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Reply by tteck** [reply]
Show `cat /etc/apt/sources.list.d/plexmediaserver.list`

**Reply by tteck** [reply]
Apparently, you don't have the repository installed. 
```
root@plex:~# apt update
Hit:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 https://downloads.plex.tv/repo/deb public InRelease
Hit:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu jammy-security InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
```

**Reply by tteck** [reply]
I'm guessing that's why you will not show the output of `cat /etc/apt/sources.list.d/plexmediaserver.list`

**Reply by tteck** [reply, marked as answer]
The issue is either due to you misconfiguring your setup or not utilizing the script provided in this repository.

**Reply by nmbrg** [reply]
i was busy with something else, sorry for not answering right away

heres the output:
```
root@plex:/# cat /etc/apt/sources.list.d/plexmediaserver.list
# deb [arch=amd64] https://downloads.plex.tv/repo/deb/ public main # disabled on upgrade to jammy
```

does it mean its disabled? how in earth is that possible? have been updating PMS without issues before and it was quite some time since i upgraded ubuntu

can i just remove the # at the beginning to enable it again?

**Reply by tteck** [reply]
you have it commented out, why?

**Reply by tteck** [reply]
yes. also remove ` # disabled on upgrade to jammy`

**Reply by nmbrg** [reply]
no idea why it is commented out. not something ive done. perhaps was done automatically when i upgraded ubuntu. but that doesnt explain how it have managed to get plex server updates installed after i upgraded ubuntu (since ive been running the apt update/upgrade command lots of times since then for updating plex server)

it didnt like me enabling it, when running apt update:
`W: https://downloads.plex.tv/repo/deb/dists/public/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.`

**Reply by tteck** [reply]
Try this:
```
rm /etc/apt/trusted.gpg.d/PlexSign.asc
wget -qO- https://downloads.plex.tv/plex-keys/PlexSign.key >/usr/share/keyrings/PlexSign.asc
echo "deb [signed-by=/usr/share/keyrings/PlexSign.asc] https://downloads.plex.tv/repo/deb/ public main" >/etc/apt/sources.list.d/plexmediaserver.list
```

**Reply by tteck** [reply]
I take it that it worked?

**Reply by nmbrg** [reply]
Sorry, i have other stuff to do as well so thats why it takes a while sometimes to answer. But thanks for sticking with me.

No, it did not work. cant remove it:
```
root@plex:/etc/apt# rm /etc/apt/trusted.gpg.d/PlexSign.asc
rm: cannot remove '/etc/apt/trusted.gpg.d/PlexSign.asc': No such file or directory
```

rm /etc/apt/trusted.gpg/PlexSign.asc
(without the .d)

**Reply by tteck** [reply]
Continue with the other 2 commands.

**Reply by nmbrg** [reply]
ok. looks like it works now. at least i didnt get the deprecation message, and plex is updated.

thanks for helping me!

**Reply by K3vb3rt** [reply]
Sorry to have to open this again. How can I get Proxmox to pull the BETA updates as well?

**Reply by Mellowlynx** [reply]
> Sorry to have to open this again. How can I get Proxmox to pull the BETA updates as well?

I use: https://github.com/mrworf/plexupdate and it's works great.

**Reply by tteck** [reply]
Use the `update` command and select option 2
https://github.com/tteck/Proxmox/blob/5f34c2c7c892da46a902853f7c1640243a2c6133/ct/plex.sh#L70-L73
